src/main/java/at/hannibal2/skyhanni/features/mining/crystalhollows/CrystalHollowsSharedLocations.kt

```diff
@@ -0,0 +1,340 @@
+            )
+        }&Key=$apiKey"
+        try {
+            val result = withContext(Dispatchers.IO) { APIUtil.getJSONResponse(url) }.asJsonObject
+            val success = result["success"].asBoolean
+            if (success) {
+                logger.log("Successful send $locationName to the server")
+                LorenzUtils.debug("CH share location $locationName at $coordinate")
+            }
+        } catch (e: Exception) {
+            CopyErrorCommand.logError(
+                Exception(
+                    "Error in addCoordinates with location: '$locationName', coordinate: '$coordinate', url: '$url'",
+                    e
+                ),
+                "Error while trying to share a ch waypoint to api.dragon99z.de"
+            )
+        }
+    }
+
+    private suspend fun removeLocation(locationName: String) {
+        if (!isEnabled()) return
+        if (!userAdded) return
+        if (!locationsNames.contains(locationName)) return
+        if (locations.any { it.name != locationName }) return
+
+        val serverId = LorenzUtils.skyBlockServerId
+        val uuid = LorenzUtils.getPlayerUuid()
+        val encodedLocation = withContext(Dispatchers.IO) {
+            URLEncoder.encode(locationName, "UTF-8")
+        }
+
+        val url =
+            "$baseAddress/api/removeLocation?serverId=$serverId&userId=$uuid&location=$encodedLocation&Key=$apiKey"
+        try {
+            val result = withContext(Dispatchers.IO) { APIUtil.getJSONResponse(url) }.asJsonObject
+            val success = result["success"].asBoolean
+            if (success) {
+                logger.log("Successful removed $locationName from the server")
+                LorenzUtils.debug("CH sharing remove location $locationName")
+            }
+        } catch (e: Exception) {
+            CopyErrorCommand.logError(
+                Exception("Error in removeLocation with location: '$locationName', url: '$url'", e),
+                "Error while trying remove a ch waypoint in api.dragon99z.de"
+            )
+        }
+    }
+
+    private fun remove(locationName: String) {
+        SkyHanniMod.coroutineScope.launch {
+            removeLocation(locationName)
+        }
+    }
+
+    private fun update(locationName: String, coordinate: LorenzVec) {
+        SkyHanniMod.coroutineScope.launch {
+            addUser()
+            fetchLocations()
+            shareLocation(locationName, coordinate.roundLocationToBlock())
+        }
+    }
+
+    @SubscribeEvent
+    fun onTimer(event: LorenzTickEvent) {
+        if (!event.isMod(40)) return
+        if (!isEnabled()) {
+            if (userAdded) {
+                SkyHanniMod.coroutineScope.launch {
+                    removeUser()
+                }
+                CrystalHollowNPCSkins.entries.forEach { it.found = false }
+                balFound = false
+            }
+            return
+        }
+        update(LorenzUtils.skyBlockArea, LocationUtils.playerLocation())
```

Comment:
Please either add an option to configure those locations, or, generate a popup that asks players if they want to share. Implicitly sharing locations isn't great, especially if some locations like the dragons lair or mines of divan, might be a thing that players sometimes dont want to share.